openpype/settings/entities/schemas/projects_schema/schema_project_ftrack.json

```diff
@@ -1114,6 +1114,142 @@
                             }
                         }
                     ]
+                },
+                {
+                    "type": "dict",
```

Comment:
Is there a reason these need to be separate settings for Farm verus Local, etc. Would it make more sense to just have a single plugin that does the Ftrack status decisions which in it has a filter setting where a profile matches only IF it's set to farm or set to local, or set to whatever. I'm a bit worried we're going to get tons of plugins as the variety of cases start to increase?

Reply:
Like I've described in PR describtion. I don't know how would that be possible with different targets, also I'm not sure how would that setting look like, it would be massive headache to handle and set all 3 cases in single plugin.

Reply:
On the plugin:
```
targets = ["local", "farm"]
```
Then in process you can just do however you like, right? (Including checking the set targets, etc.)

Reply:
How would you find out that publishing is running on farm, or locally? It does matter, we have request that if is publishing local set status A, if is being send to farm set status B, if is published on farm se status C. All 3 cases can have different logic based on when/where happens and what task type/family it is.

Reply:
I was thinking roughly like this:
```python
import pyblish.api

class Plugin(pyblish.api.InstancePlugin):
    order = pyblish.api.CollectorOrder + 0.499

    def process(self, instance):
    
        targets = pyblish.api.registered_targets()
        
        # Allow different status updates for what kind of publish
        # is currently occurring. We differentiate between a local
        # publish, a local submission for the instance to the farm
        # and a farm publish.
        if "farm" in targets:
            state = "farm"
        if "local" in targets and instance.data.get("farm"):
            state = "send_to_farm"
        else:
            # anything else we assume is just regular local
            state = "local"
            
        set_ftrack_status(instance, profiles=self.profiles, state=state)
```

Reply:
But the profiles are not the same for all states. Each state has it's own profiles.

Reply:
> Also the order of the plugins is also important, first find out which instances are going to farm, then which are local. 2 instances can lead to same task and only one status should be set. If workfile instance would be processed before render, a totally different status would be set.

Valid point.